### ISSUE TITLE
Trimmer/AOT compiler issues warnings but they are not shown and publish is successful.

### ISSUE THREAD

**Issue by voroninp** [dotnet/runtime#95932]
### Description

I see the following build output when publishing my project:
>2>Generating native code
2>"C:\Users\pavel\.nuget\packages\runtime.win-x64.microsoft.dotnet.ilcompiler\8.0.0\tools\\ilc" @"obj\Release\net8.0\win-x64\native\AotWarningsTest.ilc.rsp"
2>Assembly 'Microsoft.Build.Locator' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries
2>Assembly 'System.Composition.TypedParts' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries
2>Assembly 'Microsoft.CodeAnalysis.Workspaces' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries
2>Assembly 'System.Composition.TypedParts' produced AOT analysis warnings.
2>Assembly 'System.Composition.Hosting' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries
2>Assembly 'System.Composition.Hosting' produced AOT analysis warnings.
2>ILC: Method '[S.P.CompilerGenerated]Internal.CompilerGenerated.<Module>.ReverseDelegateStub__hostfxr_get_available_sdks_result_fn(hostfxr_resolve_sdk2_result_key_t,char**)' will always throw because: Marshaling directives are invalid
2>ILC: Method '[S.P.CompilerGenerated]Internal.CompilerGenerated.<Module>.ReverseOpenStaticDelegateStub__hostfxr_get_available_sdks_result_fn(hostfxr_resolve_sdk2_result_key_t,char**)' will always throw because: Marshaling directives are invalid
2>ILC: Method '[Microsoft.CodeAnalysis.Workspaces.MSBuild]Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.OpenSolutionAsync(string,IProgress`1<ProjectLoadProgress>,CancellationToken)' will always throw because: Failed to load assembly 'Microsoft.Build.Framework'
2>ILC: Method '[Microsoft.CodeAnalysis.Workspaces.MSBuild]Microsoft.CodeAnalysis.Host.Mef.MSBuildMefHostServices.CreateDefaultAssemblies()' will always throw because: Failed to load assembly 'Microsoft.Build.Framework'

However, I see neither warnings nor errors in the `Error List` window. The publishing is also successful. 

### Reproduction Steps

Here is the repository for the project.
It's the default project created by `Standalone Code Analysis` template.

https://github.com/voroninp/AotWarningsTest

### Expected behavior

I'd expect failure in publish and errors screaming at me.

### Actual behavior

Some hardly noticaeble mentions of Trim and AOT warnings.

**Comment by vitek-karas** [comment]
Is this publishing from VS?
If so, this is a known problem and we're working with VS to get it fixed.

**Comment by voroninp** [comment]
Yes, VS.

**Comment by voroninp** [comment]
@vitek-karas , I'd also expect `dotnet publish` to fail when treating warnings as errors. But it does not.

**Comment by vitek-karas** [comment]
> I'd also expect dotnet publish to fail when treating warnings as errors. But it does not.

That one is https://github.com/dotnet/runtime/issues/94178.
A "workaround" is to pass `/warnaserror` on the command line. I know it's not ideal...

**Comment by vitek-karas** [comment]
@voroninp I think this is now finally fixed to a place where it's usable (for .NET 9).
https://github.com/dotnet/runtime/pull/96567 implements the behavior into the compiler so that it produces errors instead of warnings.
I also verified that latest internal build of VS has the publish fixed such that warnings/errors reported by the compiler are correctly recognized as such and the Error List window is correctly populated.

The compiler fix will ship in .NET 9 Preview 2 (probably not P1). The VS side (which works just fine for warnings) should probably come up with the next preview release of VS, but I don't know exactly when it is.
